app/controllers/admin/application_controller.rb

```diff
@@ -9,8 +9,21 @@ class ApplicationController < Administrate::ApplicationController
     before_action :authenticate_user!
     before_action :authenticate_admin
 
+    # People with sysadmin privileges can do anything here. People with thesis
+    # admin privileges can create and edit but not destroy theses.
     def authenticate_admin
-      return if current_user && current_user.admin?
+      admin_actions = %w[show update index create]
+      if current_user
```

Comment:
I had trouble thinking through this logic; it's hard to read because it was hard to think through. If you have clearer ideas I would love to hear them.